crates/api_models/src/subscription.rs

```diff
@@ -140,6 +136,24 @@ pub struct PaymentDetails {
     pub payment_method_data: PaymentMethodDataRequest,
     pub setup_future_usage: Option<api_enums::FutureUsage>,
     pub customer_acceptance: Option<CustomerAcceptance>,
+    pub return_url: Option<common_utils::types::Url>,
+    pub capture_method: Option<api_enums::CaptureMethod>,
+    pub authentication_type: Option<api_enums::AuthenticationType>,
+}
+
+// Creating new type for PaymentRequest API call as usage of api_models::PaymentsRequest will result in invalid payment request during serialization
+// Eg: Amount will be serialized as { amount: {Value: 100 }}
+#[derive(Debug, Clone, serde::Serialize, ToSchema)]
+pub struct PaymentsRequestData {
+    pub payment_id: Option<common_utils::id_type::PaymentId>,
+    pub amount: Option<MinorUnit>,
+    pub currency: Option<api_enums::Currency>,
+    pub customer_id: Option<common_utils::id_type::CustomerId>,
+    pub confirm: bool,
+    pub billing: Option<Address>,
+    pub shipping: Option<Address>,
+    #[serde(flatten)]
+    pub payment_details: PaymentDetails,
```

Comment:
if `PaymentsRequestData` is introduced then what is need for having another type `PaymentDetails` and flattening the fields in the primary object, only one type is enough here